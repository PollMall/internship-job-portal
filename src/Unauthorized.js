import React from 'react';

function Unauthorized() {
  return (
    <h2>Unauthorized</h2>
  );
}

export default Unauthorized;
