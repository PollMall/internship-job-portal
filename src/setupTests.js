// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { mockApolloData } from './queries/MockApolloData';

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {},
    removeListener: () => {},
  }),
});

Object.defineProperty(window, 'alert', {
  value: jest.fn()
})

function makeMockApolloCalls(props) {
  return props.map((p) => {
    const { query, variables, data} = p;
    console.log(data);
    return {
      request: {
        query,
        variables: variables || {}
      },
      result: {
        data: data || {}
      }
    }
  })
}

const mockApolloCalls = makeMockApolloCalls(mockApolloData);

const AllProviders = ({ children }) => {
  return (
    <MockedProvider mocks={mockApolloCalls} addTypename={false}>
       <MemoryRouter>
         {children}
       </MemoryRouter>
     </MockedProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options })


export { customRender as render };