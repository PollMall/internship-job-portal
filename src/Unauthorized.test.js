import React from 'react';
import { render } from './setupTests';
import Unauthorized from './Unauthorized';

describe('Unauthorized tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<Unauthorized />);
    expect(container).toMatchSnapshot();
  });
});
