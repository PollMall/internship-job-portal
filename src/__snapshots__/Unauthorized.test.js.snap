// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized tests should match snapshot 1`] = `
<div>
  <h2>
    Unauthorized
  </h2>
</div>
`;
